plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -482,6 +484,34 @@ public void testGetTableStatisticsDoNotCacheEmptyWhenCachingMissingIsDisabled()
         this.jdbcClient.dropTable(SESSION, table);
     }
 
+    @Test
+    public void testDifferentIdentityKeys()
+    {
+        CachingJdbcClient cachingJdbcClient = new CachingJdbcClient(
+                database.getJdbcClient(),
+                SESSION_PROPERTIES_PROVIDERS,
+                new TestingJdbcIdentityCacheMapping(),
+                FOREVER,
+                true);
+        ConnectorSession alice = createSession("alice");
+        ConnectorSession bob = createSession("bob");
+
+        JdbcTableHandle table = createTable(new SchemaTableName(schema, "table"));
+
+        assertTableNamesCache(cachingJdbcClient).loads(2).misses(2).afterRunning(() -> {
+            assertThat(cachingJdbcClient.getTableNames(alice, Optional.empty())).contains(table.getRequiredNamedRelation().getSchemaTableName());
+            assertThat(cachingJdbcClient.getTableNames(bob, Optional.empty())).contains(table.getRequiredNamedRelation().getSchemaTableName());
+        });
+
+        assertTableNamesCache(cachingJdbcClient).hits(2).afterRunning(() -> {
+            assertThat(cachingJdbcClient.getTableNames(alice, Optional.empty())).contains(table.getRequiredNamedRelation().getSchemaTableName());
+            assertThat(cachingJdbcClient.getTableNames(bob, Optional.empty())).contains(table.getRequiredNamedRelation().getSchemaTableName());
+        });
+
+        // Drop tables by not using caching jdbc client
```

Comment:
I don't understand this comment

Reply:
I guess it was to mean "use non-caching jdbc client to perform cleanup"